### DISCUSSION TITLE
Rename data_description modality field to modalities

### DISCUSSION THREAD

**Discussion by dbirman** [AllenNeuralDynamics/aind-data-schema#1101] [Ideas]
It accepts a list, would be less confusing if it was plural.

This wouldn't be a backward compatible change, would need to add a before validator to swap the name in old metadata files.

**Comment by bruno-f-cruz** [comment]
This pattern has some caveats that I also went through when implementing https://github.com/AllenNeuralDynamics/aind-data-schema-models/pull/61/files that I will document here.

- It's backwards compatible from the point of view of the python library but NOT the json-schema. This is important in case people are using them for validation. (not saying its a killer but it should be contemplated).
- If we start to use this pattern more often, it may be a good idea to add a helper method/decorator for tagging features as deprecated to be removed at a later date (i.e. each time the validator is called, also emit this warning).
- The change should still be tagged as "breaking change" in the release notes

**Comment by dbirman** [comment]
Additional note here: there are many records where modality was uploaded as not a list, although I'm sure some people wouldn't pay attention the mis-named field almost certainly contributes to this

**Comment by dbirman** [comment]
Snuck this into 2.0, closing.
